### ISSUE TITLE
When ship is retreating through enemy areas, combat&exploration time is ignored

### ISSUE THREAD

**Issue by Burchov** [MaFi-Games/Captain-of-Industry-issues#1223]
### Issue description

When the ship is retreating, it always picks the shortest path back home. In the case of the attached save, you'll see that this path contains two areas that still have enemies in them.

The ship enters combat in those areas just fine, but the time the combat takes, as well as the time it takes to explore an area is ignored. You can observe this in the attached save: The area with the oil well and the one immediately to the right of it still have enemies, but they are explored for free and the two combat situations take place instantly after one another.

### Reproduction steps

1) Open the attached save. Here I ordered the ship to attack the area west of an oil rig.
2) Look at the map: you will see the ship traveling to an area occupied by enemy forces.
3) Wait a bit. The ship will enter combat at the first site.
4) After combat, the ship automatically retreats to the oil rig, initiating combat there
5) After the combat at the oil rig is finished, combat is immediately initiated again, because the ship somehow already explored the area to the right of it (between the oil rig and the already explored limestone quarry). 

I would expect that at step 5 the combat for the second site is only initiated after 1) the combat for the oil rig is over, 2) after the ship travels there and 3) after the ship explored the area

### Game version and build number

Update 1 | v0.5.3e (b225)

### Operating system

Windows

### Attachments

[Ship bugreport.zip](https://github.com/MaFi-Games/Captain-of-Industry-issues/files/12270056/Ship.bugreport.zip)

**Comment by JamieRCHI** [comment]
I agree. The return path should be remembered and used. Maybe ignore fuel requirements when retreating?
